src/power_hand.c

```diff
@@ -201,10 +200,32 @@ long can_thing_be_picked_up_by_player(const struct Thing *thing, PlayerNumber pl
     return can_cast_spell(plyr_idx, PwrK_HAND, thing->mappos.x.stl.num, thing->mappos.y.stl.num, thing, CastChk_Default);
 }
 
-long can_thing_be_picked_up2_by_player(const struct Thing *thing, PlayerNumber plyr_idx)
+TbBool can_thing_be_picked_up2_by_player(const struct Thing *thing, PlayerNumber plyr_idx)
 {
     //TODO: rewrite, then give it better name
-    return _DK_can_thing_be_picked_up2_by_player(thing, plyr_idx);
+    // return _DK_can_thing_be_picked_up2_by_player(thing, plyr_idx);
+    if ( thing->class_id != TCls_Creature )
+    {
+        return ( (thing->class_id == TCls_Object) && (object_is_pickable_by_hand_for_use(thing, plyr_idx)) );
+    }
+    if ( (game.armageddon_cast_turn > 0) && ( ((game.armageddon.count_down + game.armageddon_cast_turn) <= game.play_gameturn) ) )
+    {
+        return false;
+    }
+    if ( (thing->active_state == CrSt_CreatureUnconscious) || ((thing->alloc_flags & TAlF_IsInLimbo) != 0) || ((thing->state_flags & TF1_InCtrldLimbo) != 0) || (thing->health <= 0) )
+    {
+        return false;
+    }
+    unsigned char state = (thing->active_state == CrSt_MoveToPosition) ? thing->continue_state : thing->active_state;
```

Comment:
I'm sure this code is incorrect, but such a notation is very hard to read and understand for humans. Could you write it out with 'if'-statements?